internal/util/version.go

```diff
@@ -1,57 +1,92 @@
 package util
 
 import (
+	"fmt"
 	"path/filepath"
-	"strings"
 
 	"github.com/edulinq/autograder/internal/log"
 )
 
 const (
 	UNKNOWN_VERSION  string = "???"
 	UNKNOWN_HASH     string = "????????"
-	VERSION_FILENAME string = "VERSION.txt"
+	VERSION_FILENAME string = "VERSION.json"
 	DIRTY_SUFFIX     string = "dirty"
+	UNKNOWN_API      int    = 0
 	HASH_LENGTH      int    = 8
 )
 
-func GetAutograderVersion() string {
+type Version struct {
+	Short   string `json:"short-version"`
+	Hash    string `json:"git-hash"`
+	IsDirty bool   `json:"is-dirty"`
+	Api     int    `json:"api-version"`
+}
+
+func (this Version) String() string {
+	fullVersion := this.Short + "-" + this.Hash
+	if this.IsDirty {
+		fullVersion = fullVersion + "-" + DIRTY_SUFFIX
+	}
+
+	return fullVersion
+}
+
+func MustGetAPIVersion() int {
+	version, err := versionFromJSON()
+	if err != nil {
+		log.Fatal("Failed to read the version from JSON file.", err)
+	}
+
+	return version.Api
+}
+
+func versionFromJSON() (Version, error) {
+	var version Version
```

Comment:
You are missing the initialization of this (you don't want the default empty value, you want a custom empty/default value).